### PR TITLE
enh(swift) regex literal support

### PR DESCRIPTION
**Pull request by bradleymackey**
Swift 5.7 introduced regex literals, specified in [SE-0354](https://github.com/apple/swift-evolution/blob/main/proposals/0354-regex-literals.md).

This includes support for bare-slash (`/example/`) and extended regex literals (`#/example/#`, `##/example/##`, etc).
- Bare slash literals are not enabled by default in Swift 5, but will be in Swift 6.
- Bare slash literals cannot start with whitespace, but extended literals can.
- Only extended regex literals support newlines.
- Multiline extended literals support comments starting with `#`, continuing until end of line.
- Follows convention of raw strings, supporting up to 3 levels of delimiting for extended literals, ensuring `#` delimiters match.

### Changes
- Add regex literal support

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by bradleymackey** [comment]
Closing because of edge cases with bare-slash literals and operators. I'll reopen once resolved.

File: src/languages/swift.js [hunk 1]
```diff
@@ -180,6 +180,54 @@ export default function(hljs) {
...
+    //
+    //     3 / something
+    //
+    // (which will then blow up when regex's `illegal` sees the newline)
+    begin: /(?=\/[^\s][^/\n]*\/)/,
+    contains: [
+      {
+        begin: /\/[^\s]/,
+        end: /\//,
+        illegal: /\n/,
```
Comment (joshgoebel) [inline src/languages/swift.js:207]: Is this needed with the "lookahead" you're doing on line 202 which should guarantee there will be no newlines?

Comment (bradleymackey) [reply src/languages/swift.js:207]: You're right, this works as intended without this additional rule. This was heavily inspired by `REGEXP_MODE` which uses the [same `illegal` rule](https://github.com/highlightjs/highlight.js/blob/98be55bf6520f62c581a498509baf9387461dce0/src/lib/modes.js#L164). The suite also passes when this `illegal` rule is removed as well, should I include that removal in this patch as well for consistency?

**Comment by bradleymackey** [comment]
Added comment support for extended literals, starting with `#` ending with a newline.

File: src/languages/swift.js (hunk 1 above)
Comment (joshgoebel) [reply src/languages/swift.js]: Sure, I can't see what purpose it's serving.  I wonder if the whole rule also couldn't be flattened by just lifting the rule out of the contains and adding the lookahead to it's begin?

Comment (bradleymackey) [reply src/languages/swift.js]: Good suggestion! Yes did the trick much more elegantly

**Comment by bradleymackey** [comment]
Fixed comments in extended literals, only multiline literals support comments.
